src/components/ProductDetailAddToCart/ProductDetailAddToCart.js

```diff
@@ -0,0 +1,204 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { withStyles } from "material-ui/styles";
+import ButtonBase from "material-ui/ButtonBase";
+import Grid from "material-ui/Grid";
+import InputAdornment from "material-ui/Input/InputAdornment";
+import TextField from "material-ui/TextField";
+import Typography from "material-ui/Typography";
+import Minus from "mdi-material-ui/Minus";
+import Plus from "mdi-material-ui/Plus";
+import { inject, observer } from "mobx-react";
+import Divider from "components/Divider";
+
+const styles = (theme) => ({
+  addToCartButton: {
+    "padding": theme.spacing.unit,
+    "backgroundColor": theme.palette.primary.light,
+    "borderRadius": theme.palette.reaction.buttonBorderRadius,
+    "minWidth": "66%",
+    "&:hover": {
+      borderColor: theme.palette.reaction.activeElementBorderColor
+    },
+    "&:focus": {
+      outline: "auto 5px -webkit-focus-ring-color"
+    }
+  },
+  addToCartText: {
+    color: theme.palette.primary.contrastText,
+    fontWeight: 600
+  },
+  incrementButton: {
+    backgroundColor: theme.palette.reaction.black02,
+    color: theme.palette.reaction.coolGray500,
+    fontSize: "12px",
+    padding: 6
+  },
+  quantityContainer: {
+    padding: 0,
+    border: `1px solid ${theme.palette.reaction.black15}`,
+    backgroundColor: theme.palette.common.white,
+    borderRadius: theme.palette.reaction.buttonBorderRadius
+  },
+  quantityGrid: {
+    marginBottom: theme.spacing.unit * 3
+  },
+  quantityInput: {
+    "color": theme.palette.reaction.coolGray500,
+    "fontSize": "12px",
+    "width": "40px",
+    "textAlign": "center",
+    "&:focus": {
+      borderColor: "#80bdff",
+      boxShadow: "0 0 0 0.2rem rgba(0,123,255,.25)"
+    },
+    "borderLeft": `1px solid ${theme.palette.reaction.black15}`,
+    "borderRight": `1px solid ${theme.palette.reaction.black15}`
+  },
+  quantityTypography: {
+    color: theme.palette.reaction.coolGray500,
+    marginBottom: theme.spacing.unit * 2
+  }
+});
+
+
+@withStyles(styles)
+@inject("uiStore")
+@observer
+export default class ProductDetailAddToCart extends Component {
+  static propTypes = {
+    classes: PropTypes.object,
+    onClick: PropTypes.func.isRequired,
+    option: PropTypes.object,
+    selectedOption: PropTypes.string
+  }
+
+  static propTypes = {
+    classes: PropTypes.object,
+    uiStore: PropTypes.object
+  };
+
+  static defaultProps = {
+    classes: {},
+    uiStore: {}
+  };
+
+  state = {
+    addToCartQuantity: 1
+  };
+
+  handleOnClick = () => {
+    // This function currently does nothing. When our GraphQL endpoints are available, we'll use them to add the items to the cart.
```

Comment:
@aldeed this looks like a good place to elevate state management logic to a higher level, perhaps to the page level, and remove the mobx dependency for this component. A prop could be called here instead. Thoughts?

Reply:
Good point. @kieckhafer The parent component, ProductDetail, has uiState, so you can move this code there and you won't need to inject state here.

So at top of the parent render function you would move:

```js
const { uiStore: { pdpSelectedOptionId, pdpSelectedVariantId } } = this.props;

let pdpProductToAddToCart = pdpSelectedVariantId;
if (pdpSelectedOptionId) {
  pdpProductToAddToCart = pdpSelectedOptionId;
}
```

And then do:

```js
<ProductDetailAddToCart variantId={pdpProductToAddToCart} />
```

And change this code here to use the passed in `variantId` prop.